src/psyclone/tests/psyGen_test.py

```diff
@@ -2274,3 +2274,87 @@ def test_kern_ast():
     assert isinstance(kern, GOKern)
     assert kern.ast
     assert isinstance(kern.ast, Fortran2003.Program)
+
+
+def test_dataaccess_vector():
+    '''Test that the DataAccess class works as expected when we have a
+    vector field argument that depends on more than one halo exchange
+    (due to halo exchanges working separately on vectors)
+
+    '''
+    _, invoke_info = parse(
+        os.path.join(BASE_PATH, "4.9_named_multikernel_invokes.f90"),
+        distributed_memory=True, api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3",
+                     distributed_memory=True).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+
+    # d from halo exchange vector 1
+    halo_exchange_d_v1 = schedule.children[3]
+    field_d_v1 = halo_exchange_d_v1.field
+    # d from halo exchange vector 2
+    halo_exchange_d_v2 = schedule.children[4]
+    field_d_v2 = halo_exchange_d_v2.field
+    # d from halo exchange vector 3
+    halo_exchange_d_v3 = schedule.children[5]
+    field_d_v3 = halo_exchange_d_v3.field
+    # d from a kernel argument
+    loop = schedule.children[6]
+    kernel = loop.children[0]
+    d_arg = kernel.arguments.args[5]
+
+    access = DataAccess(d_arg)
+    assert not access.covered
+
+    access.update_coverage(field_d_v3)
+    assert not access.covered
+    access.update_coverage(field_d_v2)
+    assert not access.covered
+
+    with pytest.raises(InternalError) as excinfo:
+        access.update_coverage(field_d_v3)
+    assert (
+        "Found more than one dependent halo exchange with the same vector "
+        "index" in str(excinfo.value))
+
+    access.update_coverage(field_d_v1)
+    assert access.covered
+
+    access.reset_coverage()
+    assert not access.covered
+    assert not access._vector_index_access
+
+
+def test_dataaccess_vector_size_differs(monkeypatch):
+    '''If update_coverage() is called from DataAccess and the arguments
+    are the same vector field, then the field vector sizes should be
+    the same. This test checks that an exception is raised if they
```

Comment:
I don't see an exception being raised about the differing vector lengths. Only one for the sanity check on the fact that the vector indices are the same.

Reply:
You are right. I'd made the previous test always return True so any two halo exchange vectors would trip this exception. I've now updated the test.